src/main/java/com/educare/unitylend/controller/BorrowRequestController.java

```diff
@@ -57,58 +95,73 @@ ResponseEntity<List<BorrowRequest>> getBorrowRequestForUserId(@PathVariable Stri
      */
     @GetMapping("/get-borrow-request-by-community-id/{communityId}")
     ResponseEntity<List<BorrowRequest>> getBorrowRequestForCommunity(@PathVariable String communityId) throws ControllerException {
-        return null;
+        try {
+            if (communityId == null || communityId.isEmpty()) {
+                log.error("Community id cannot be null");
+                return ResponseEntity.badRequest().body(null);
+            }
+            List<BorrowRequest> borrowRequestList = borrowRequestCommunityMapService.getRequestsByCommunityId(communityId);
+            if (borrowRequestList == null || borrowRequestList.isEmpty()) {
+                log.error("No borrow requests found for given community");
+                return ResponseEntity.badRequest().body(null);
+            }
+            return ResponseEntity.ok(borrowRequestList);
+        } catch (ServiceException e) {
+            log.error("Error encountered in fetching borrow requests in a community", e);
+            throw new ControllerException("Error encountered in fetching borrow requests in a community", e);
+        }
     }
 
-
     /**
      * API endpoint for retrieving borrow requests in a community within an amount range.
-     *
+     *@PathVariable String communityId
      * @param minAmount   The minimum amount of the borrow requests.
      * @param maxAmount   The maximum amount of the borrow requests.
      * @param lessThan    Flag indicating if the amount should be less than maxAmount.
      * @param greaterThan Flag indicating if the amount should be greater than minAmount.
      * @return ResponseEntity<List<BorrowRequest>> The list of borrow requests in the community within the specified amount range.
      * @throws ControllerException If an error occurs during the borrow request retrieval process.
      */
-    @GetMapping("/get-borrow-request-in-community-by-amount")
+    @PostMapping("/min/{communityId}")
     ResponseEntity<List<BorrowRequest>> getBorrowRequestsInCommunityInAmountRange(
             @RequestParam(value = "minAmount", required = false) BigDecimal minAmount,
-            @RequestParam(value = "maxAmount", required = false) BigDecimal maxAmount,
+            @RequestParam(value = "maxAmount", required = true) BigDecimal maxAmount,
             @RequestParam(value = "lessThan", required = false, defaultValue = "false") boolean lessThan,
-            @RequestParam(value = "greaterThan", required = false, defaultValue = "true") boolean greaterThan
+            @RequestParam(value = "greaterThan", required = false, defaultValue = "true") boolean greaterThan,
+            @PathVariable String communityId
     ) throws ControllerException {
```

Comment:
Can the above function be changed to have just 2 parameters along with communityId (lowerLimit and upperLimit)?